cynic-codegen/src/fragment_derive/input.rs

```diff
@@ -17,6 +17,24 @@ pub struct FragmentDeriveInput {
 
 impl FragmentDeriveInput {
     pub fn validate(&self) -> Result<(), Errors> {
+        let data_field_is_empty = self
+            .data
+            .clone()
+            .map_struct_fields(|field| field.ident)
+            .take_struct()
+            .unwrap()
+            .fields
+            .is_empty();
```

Comment:
Think you can probably simplify this:

```suggestion
let data_field_is_empty = matches!(self.data, darling::ast::Data::Struct(fields) if fields.fields.is_empty());
```

Reply:
Good way 👍